src/FileCacheManager.php

```diff
@@ -86,15 +86,22 @@ public function needFixing($file, $fileContent)
         return false;
     }
 
-    public function setFile($file, $fileContent)
+    public function setFile($file, $fileContent, $isDryRun)
     {
         if (!$this->isCacheAvailable()) {
             return;
         }
 
         $file = $this->getRelativePathname($file);
 
-        $this->newHashes[$file] = $this->calcHash($fileContent);
+        $hash = $this->calcHash($fileContent);
+
+        // set hash to an impossible value in dry-run so that the file will definitely be inspected on the next run
+        if ($isDryRun && array_key_exists($file, $this->newHashes) && $this->newHashes[$file] !== $hash) {
+            $hash = -1;
```

Comment:
@keradus @stof 

Instead of setting to the hash to `-1`, we can of course set to something else, for example `null` or `false`.

What do you think?